stdlib/public/Concurrency/Task.swift

```diff
@@ -323,6 +323,21 @@ func isTaskCancelled(_ task: Builtin.NativeObject) -> Bool
 @_silgen_name("swift_task_runAndBlockThread")
 public func runAsyncAndBlock(_ asyncFun: @escaping () async -> ())
 
+@_silgen_name("swift_task_asyncMainDrainQueue")
+public func _asyncMainDrainQueue() -> Never
+
+public func _runAsyncMain(_ asyncFun: @escaping () async throws -> ()) {
```

Comment:
`_runAsyncMain` should return `Never`?

Reply:
Yes, probably. I'll have to hunt down where we type check main though since the type checker gets mad when main doesn't return `void`. Just leaving this `void` and letting the `_asyncMainDrainQueue` return `Never` seems to work. Thoughts?